projects/observability/src/shared/services/csv-downloader/csv-downloader.service.ts

```diff
@@ -0,0 +1,39 @@
+import { Injectable } from '@angular/core';
+import { Dictionary } from '@hypertrace/common';
+import { CsvDownloadFileConfig, FileDownloadService } from '@hypertrace/components';
+import { Observable, of } from 'rxjs';
+
+@Injectable({ providedIn: 'root' })
+export class CSVDownloaderService {
+  private readonly componentHandlers: Map<string, Observable<Dictionary<unknown>[]>> = new Map();
+
+  public constructor(private readonly fileDownloaderService: FileDownloadService) {}
+
+  public downloadCsv(key: string, fileDownloadConfig?: CsvDownloadFileConfig): void {
```

Comment:
1. You can a partial type of the `CsvDownloadFileConfig` to get the data in the function instead of `undefined`
2. Should we through an error if the data source is not registered? I think we can try that.

Reply:
`done`